sssd_test_framework/utils/local_override.py

```diff
@@ -0,0 +1,445 @@
+    """
+
+    def __init__(self, host: MultihostHost) -> None:
+        """
+        :rtype: object
+        :param host: Remote host instance.
+        :type host: MultihostHost
+        """
+        super().__init__(host)
+
+        self.cli: CLIBuilder = CLIBuilder(host.ssh)
+        self._users: list[str] = []
+        self._groups: list[str] = []
+
+    def teardown(self) -> None:
+        """
+        Teardown
+        """
+        cmd = ""
+        if self._users:
+            cmd += "\n".join([f"sss_override user-del '{x}'" for x in self._users])
+            cmd += "\n"
+
+        if self._groups:
+            cmd += "\n".join([f"sss_override group-del '{x}'" for x in self._groups])
+            cmd += "\n"
+
+        if cmd:
+            self.host.ssh.run(cmd, raise_on_error=False)
+
+        super().teardown()
+
+    def user(self, name: str) -> LocalOverrideUser:
+        """
+        Get local override user object.
+
+        .. code-block:: python
+            :caption: Example usage
+
+            @pytest.mark.topology(KnownTopology.Client)
+            def test_example(client: Client):
+                # Create override
+                client.local_override.users("user1").add(name="o-user1", uid=10001, gid=10001, gecos="test")
+
+                # Assert the overridden name, uid, gid and gecos
+                assert client.tools.getent.passwd("o-user1").uid == 10001
+                assert client.tools.getent.passwd("o-user1").gid == 10001
+                assert client.tools.getent.passwd("o-user1").gecos == "test"
+
+        :param name: User.
+        :type name: str
+        :return: New user local override object.
+        :rtype: LocalOverrideUser
+        """
+        return LocalOverrideUser(self, name)
+
+    def group(self, group: str) -> LocalOverrideGroup:
+        """
+        Get local override group object.
+
+        .. code-block:: python
+            :caption: Example usage
+
+            @pytest.mark.topology(KnownTopology.Client)
+            def test_example(client: Client):
+                # Create local override
+                client.local_override.group("users").add(name="o-users", gid=10001)
+
+                # Assert the overridden name and gid
+                assert client.tools.getent.group("o-users").gid == 10001
+
+        :param group: Group name.
+        :type group: str
+        :return: New group object.
+        :rtype: LocalOverrideGroup
+        """
+        return LocalOverrideGroup(self, group)
+
+    @property
+    def users(self):
+        return self._users
+
+
+class LocalOverrideUser:
+    """
+    Management of local override users.
+    """
+
+    def __init__(self, util: LocalOverrideUtils, user: str) -> None:
+        """
+        :param util: OverrideUser object.
+        :type util: LocalOverrideUser
+        :param user: User name.
+        :type user: str
+        """
+        self.util = util
+        self.user = user
```

Comment:
Name is an extra argument, it's not the same as user.

Reply:
What do you mean?

Reply:
The command is this,

sss_override user-add pbrezina --uid 1234 --gid 1234 --name pavel 

We can change it to username?

Reply:
Ah, I see. But it is fine, `self.name` and `name` are two different things. It's perfectly fine to use both, although it may create little confusion, it is better to solve the confusion by making a comment where you introduce local `name`.

If you use `self.username` and `name` or the other way around, imho the confusion is still there (ie which name is which). You could solve it by using `override_name` but that is too long to use nicely.